### PR TITLE
Manage abilities proxy

### PR DESCRIPTION
**Pull request by MoMannn**
TODO:

- [x] setAbilities
- [x] orderGateway action for setting abilities

### CODE REVIEW

File: docs/api/ethereum.md [hunk 1]
```diff
@@ -2456,6 +2456,51 @@ const mutation = await ledger.revokeAsset(assetId);
 
 [destroyAsset](#destroyasset-assetid)
 
+### setAbilities(accountId, abilities)
+
+An `asynchronous` class instance `function` which sets `abilities` of an `accountId`.
+
+::: warning
```
Comment (typwrtr) [inline docs/api/ethereum.md]: ::: Warning

File: docs/api/ethereum.md [hunk 2]
```diff
@@ -2456,6 +2456,51 @@ const mutation = await ledger.revokeAsset(assetId);
...
+The `MANAGE_ABILITIES` super ability of the ledger is required to perform this function.
+:::
+
+::: warning
```
Comment (typwrtr) [inline docs/api/ethereum.md]: ::: Warning

File: docs/api/ethereum.md [hunk 3]
```diff
@@ -2569,18 +2614,19 @@ This categorization is for safety purposes since revoking your own super ability
 | Name | Value | Description
 |-|-|-
 | MANAGE_ABILITIES | 1 | Allows an account to further grant abilities.
+| ALLOW_MANAGE_ABILITIES | 2 | A specific ability that is bounded to atomic orders. When graning or revoking abilitites trough `Gateway` the order maker has to have this ability.
```
Comment (typwrtr) [inline docs/api/ethereum.md]: | ALLOW_MANAGE_ABILITIES | 2 | A specific ability that is bounded to atomic orders. When granting or revoking abilities through `Gateway`, the order maker has to have this ability.

File: docs/api/ethereum.md [hunk 4]
```diff
@@ -2569,18 +2614,19 @@ This categorization is for safety purposes since revoking your own super ability
...
 | Name | Value | Description
 |-|-|-
-| ALLOW_CREATE_ASSET | 32 | A specific ability that is bounded to atomic orders. When creating a new asset trough `Gateway`, the order maker has to have this ability.
-| ALLOW_UPDATE_ASSET_IMPRINT | 128 | A specific ability that is bounded to atomic orders. When updating asset imprint trough `Gateway`, the order maker has to have this ability.
-| CREATE_ASSET | 2 | Allows an account to create a new asset.
-| REVOKE_ASSET | 4 | Allows management accounts to revoke assets.
-| TOGGLE_TRANSFERS | 8 | Allows an account to stop and start asset transfers.
-| UPDATE_ASSET | 16 | Allows an account to update asset data.
-| UPDATE_URI | 64 | Allows an account to update asset ledger's URI prefix and postfix.
+| ALLOW_CREATE_ASSET | 512 | A specific ability that is bounded to atomic orders. When creating a new asset trough `Gateway`, the order maker has to have this ability.
```
Comment (typwrtr) [inline docs/api/ethereum.md]: | ALLOW_CREATE_ASSET | 512 | A specific ability that is bounded to atomic orders. When creating a new asset through `Gateway`, the order maker has to have this ability.

File: docs/api/ethereum.md [hunk 5]
```diff
@@ -2569,18 +2614,19 @@ This categorization is for safety purposes since revoking your own super ability
...
+| ALLOW_UPDATE_ASSET_IMPRINT | 1024 | A specific ability that is bounded to atomic orders. When updating asset imprint trough `Gateway`, the order maker has to have this ability.
```
Comment (typwrtr) [inline docs/api/ethereum.md]: | ALLOW_UPDATE_ASSET_IMPRINT | 1024 | A specific ability that is bounded to atomic orders. When updating asset imprint through `Gateway`, the order maker has to have this ability.

File: docs/api/ethereum.md [hunk 6]
```diff
@@ -3203,13 +3249,18 @@ Multi-order actions define the atomic operations of the actions order.
 |-|-|-
 | CREATE_ASSET | 1 | Create a new asset.
 | UPDATE_ASSET_IMPRINT | 4 | Update asset imprint.
+| SET_ABILITIES | 5 | Sets abilities.
```
Comment (typwrtr) [inline docs/api/ethereum.md]: | SET_ABILITIES | 5 | Set abilities.

File: docs/api/ethereum.md [hunk 7]
```diff
@@ -3203,13 +3249,18 @@ Multi-order actions define the atomic operations of the actions order.
...
 | TRANSFER_ASSET | 2 | Transfer an asset.
 | TRANSFER_VALUE | 3 | Transfer a value.
 
 ::: Warning
 There is a possibility of unintentional behavior where asset imprint can be overwritten if more than one `UPDATE_ASSET_IMPRINT` order per asset is active. Be aware of this when implementing.
 :::
 
+::: Warning
+There is a possibility of unintentional behavior where account abilitites can be overwritten if more than one `SET_ABILITIES` order per account is active. Be aware of this when implementing.
```
Comment (typwrtr) [inline docs/api/ethereum.md]: There is a possibility of unintentional behavior where account abilities could be overwritten if more than one `SET_ABILITIES` order per account is active. Be aware of this when implementing.

File: docs/api/wanchain.md [hunk 8]
```diff
@@ -21,9 +21,9 @@ A `class` providing communication with the Wanchain blockchain using the HTTP/HT
 | options.cache | A `string` representing request cache type. It defaults to `no-cache`. Please see more details [here](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch).
 | options.credentials | A `string` representing request credentials. It defaults to `omit`. Please see more details [here](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch).
 | options.gasPriceMultiplier | A `number` represents a multiplier of the current gas price when performing a mutation. It defaults to `1.1`.
-| options.gatewayConfig.assetLedgerDeployOrderId | A `string` representing an Ethereum address of the [asset ledger deploy gateway](/#public-addresses).
-| options.gatewayConfig.actionsOrderId | A `string` representing an Ethereum address of the [actions order gateway](/#public-addresses).
-| options.gatewayConfig.valueLedgerDeployOrderId | A `string` representing an Ethereum address of the [value ledger deploy gateway](/#public-addresses).
+| options.gatewayConfig.assetLedgerDeployOrderId | A `string` representing an Wanchain address of the [asset ledger deploy gateway](/#public-addresses).
```
Comment (typwrtr) [inline docs/api/wanchain.md]: | options.gatewayConfig.assetLedgerDeployOrderId | A `string` representing a Wanchain address of the [asset ledger deploy gateway](/#public-addresses).

File: docs/api/wanchain.md [hunk 9]
```diff
@@ -21,9 +21,9 @@ A `class` providing communication with the Wanchain blockchain using the HTTP/HT
...
+| options.gatewayConfig.actionsOrderId | A `string` representing an Wanchain address of the [actions order gateway](/#public-addresses).
```
Comment (typwrtr) [inline docs/api/wanchain.md]: | options.gatewayConfig.actionsOrderId | A `string` representing a Wanchain address of the [actions order gateway](/#public-addresses).

File: docs/api/wanchain.md [hunk 10]
```diff
@@ -21,9 +21,9 @@ A `class` providing communication with the Wanchain blockchain using the HTTP/HT
...
+| options.gatewayConfig.valueLedgerDeployOrderId | A `string` representing an Wanchain address of the [value ledger deploy gateway](/#public-addresses).
```
Comment (typwrtr) [inline docs/api/wanchain.md]: | options.gatewayConfig.valueLedgerDeployOrderId | A `string` representing a Wanchain address of the [value ledger deploy gateway](/#public-addresses).

File: docs/api/wanchain.md [hunk 11]
```diff
@@ -71,9 +71,9 @@ A class instance `variable` holding a `GatewayConfig` which represents the confi
 
 | Argument | Description
 |-|-
-| assetLedgerDeployOrderId | A `string` representing an Ethereum address of the [asset ledger deploy gateway](/#public-addresses).
-| actionsOrderId | A `string` representing an Ethereum address of the [actions order gateway](/#public-addresses).
-| valueLedgerDeployOrderId | A `string` representing an Ethereum address of the [value ledger deploy gateway](/#public-addresses).
+| assetLedgerDeployOrderId | A `string` representing an Wanchain address of the [asset ledger deploy gateway](/#public-addresses).
```
Comment (typwrtr) [inline docs/api/wanchain.md]: | assetLedgerDeployOrderId | A `string` representing a Wanchain address of the [asset ledger deploy gateway](/#public-addresses).

File: docs/api/wanchain.md [hunk 12]
```diff
@@ -71,9 +71,9 @@ A class instance `variable` holding a `GatewayConfig` which represents the confi
...
+| actionsOrderId | A `string` representing an Wanchain address of the [actions order gateway](/#public-addresses).
+| valueLedgerDeployOrderId | A `string` representing an Wanchain address of the [value ledger deploy gateway](/#public-addresses).
```
Comment (typwrtr) [inline docs/api/wanchain.md]: | valueLedgerDeployOrderId | A `string` representing a Wanchain address of the [value ledger deploy gateway](/#public-addresses).

File: docs/api/wanchain.md [hunk 13]
```diff
@@ -71,9 +71,9 @@ A class instance `variable` holding a `GatewayConfig` which represents the confi
 
 | Argument | Description
 |-|-
-| assetLedgerDeployOrderId | A `string` representing an Ethereum address of the [asset ledger deploy gateway](/#public-addresses).
-| actionsOrderId | A `string` representing an Ethereum address of the [actions order gateway](/#public-addresses).
-| valueLedgerDeployOrderId | A `string` representing an Ethereum address of the [value ledger deploy gateway](/#public-addresses).
+| assetLedgerDeployOrderId | A `string` representing an Wanchain address of the [asset ledger deploy gateway](/#public-addresses).
+| actionsOrderId | A `string` representing an Wanchain address of the [actions order gateway](/#public-addresses).
```
Comment (typwrtr) [inline docs/api/wanchain.md]: | actionsOrderId | A `string` representing a Wanchain address of the [actions order gateway](/#public-addresses).

File: docs/api/wanchain.md [hunk 14]
```diff
@@ -1364,6 +1364,51 @@ const mutation = await ledger.revokeAsset(assetId);
 
 [destroyAsset](#destroyasset-assetid)
 
+### setAbilities(accountId, abilities)
+
+An `asynchronous` class instance `function` which sets `abilities` of an `accountId`.
+
+::: warning
```
Comment (typwrtr) [inline docs/api/wanchain.md]: ::: Warning

File: docs/api/wanchain.md [hunk 15]
```diff
@@ -1364,6 +1364,51 @@ const mutation = await ledger.revokeAsset(assetId);
...
+The `MANAGE_ABILITIES` super ability of the ledger is required to perform this function.
+:::
+
+::: warning
```
Comment (typwrtr) [inline docs/api/wanchain.md]: ::: Warning

File: docs/api/wanchain.md [hunk 16]
```diff
@@ -1477,17 +1522,19 @@ This categorization is for safety purposes since revoking your own super ability
 | Name | Value | Description
 |-|-|-
 | MANAGE_ABILITIES | 1 | Allows an account to further grant abilities.
+| ALLOW_MANAGE_ABILITIES | 2 | A specific ability that is bounded to atomic orders. When graning or revoking abilitites trough `Gateway` the order maker has to have this ability.
```
Comment (typwrtr) [inline docs/api/wanchain.md]: | ALLOW_MANAGE_ABILITIES | 2 | A specific ability that is bounded to atomic orders. When granting or revoking abilities through `Gateway`, the order maker has to have this ability.

File: docs/api/wanchain.md [hunk 17]
```diff
@@ -1477,17 +1522,19 @@ This categorization is for safety purposes since revoking your own super ability
...
 
 | Name | Value | Description
 |-|-|-
-| ALLOW_CREATE_ASSET | 32 | A specific ability that is bounded to atomic orders. When creating a new asset trough `Gateway`, the order maker has to have this ability.
-| CREATE_ASSET | 2 | Allows an account to create a new asset.
-| REVOKE_ASSET | 4 | Allows management accounts to revoke assets.
-| TOGGLE_TRANSFERS | 8 | Allows an account to stop and start asset transfers.
-| UPDATE_ASSET | 16 | Allows an account to update asset data.
-| UPDATE_URI | 64 | Allows an account to update asset ledger's URI prefix and postfix.
+| ALLOW_CREATE_ASSET | 512 | A specific ability that is bounded to atomic orders. When creating a new asset trough `Gateway`, the order maker has to have this ability.
```
Comment (typwrtr) [inline docs/api/wanchain.md]: | ALLOW_CREATE_ASSET | 512 | A specific ability that is bounded to atomic orders. When creating a new asset through `Gateway`, the order maker has to have this ability.

File: docs/api/wanchain.md [hunk 18]
```diff
@@ -1477,17 +1522,19 @@ This categorization is for safety purposes since revoking your own super ability
...
+| ALLOW_UPDATE_ASSET_IMPRINT | 1024 | A specific ability that is bounded to atomic orders. When updating asset imprint trough `Gateway`, the order maker has to have this ability.
```
Comment (typwrtr) [inline docs/api/wanchain.md]: | ALLOW_UPDATE_ASSET_IMPRINT | 1024 | A specific ability that is bounded to atomic orders. When updating asset imprint through `Gateway`, the order maker has to have this ability.

File: docs/api/wanchain.md [hunk 19]
```diff
@@ -1860,25 +1907,25 @@ When using dynamic order, you cannot send any of the assets to the zero address
...
 
 **Arguments**
 
 | Argument | Description
 |-|-
-| gatewayConfig.assetLedgerDeployOrderId | A `string` representing an Ethereum address of the [asset ledger deploy gateway](/#public-addresses).
-| gatewayConfig.actionsOrderId | A `string` representing an Ethereum address of the [actions order gateway](/#public-addresses).
-| gatewayConfig.valueLedgerDeployOrderId | A `string` representing an Ethereum address of the [value ledger deploy gateway](/#public-addresses).
-| provider | [required] An instance of an HTTP or MetaMask provider.
+| gatewayConfig.assetLedgerDeployOrderId | A `string` representing an Wanchain address of the [asset ledger deploy gateway](/#public-addresses).
```
Comment (typwrtr) [inline docs/api/wanchain.md]: | gatewayConfig.assetLedgerDeployOrderId | A `string` representing a Wanchain address of the [asset ledger deploy gateway](/#public-addresses).

File: docs/api/wanchain.md [hunk 20]
```diff
@@ -1860,25 +1907,25 @@ When using dynamic order, you cannot send any of the assets to the zero address
...
+| gatewayConfig.actionsOrderId | A `string` representing an Wanchain address of the [actions order gateway](/#public-addresses).
```
Comment (typwrtr) [inline docs/api/wanchain.md]: | gatewayConfig.actionsOrderId | A `string` representing a Wanchain address of the [actions order gateway](/#public-addresses).

File: docs/api/wanchain.md [hunk 21]
```diff
@@ -1860,25 +1907,25 @@ When using dynamic order, you cannot send any of the assets to the zero address
...
+| gatewayConfig.valueLedgerDeployOrderId | A `string` representing an Wanchain address of the [value ledger deploy gateway](/#public-addresses).
```
Comment (typwrtr) [inline docs/api/wanchain.md]: | gatewayConfig.valueLedgerDeployOrderId | A `string` representing a Wanchain address of the [value ledger deploy gateway](/#public-addresses).

File: docs/api/wanchain.md [hunk 22]
```diff
@@ -2056,16 +2103,16 @@ This order kind is used for delegating `AssetLedger` deploy.
...
-| assetLedgerData.owner | [required] A `string` representing Ethereum wallet, which will be the owner of the asset ledger.
+| assetLedgerData.owner | [required] A `string` representing Wanchain wallet, which will be the owner of the asset ledger.
 | assetLedgerData.schemaId | [required] A `string` representing data schema ID.
 | assetLedgerData.symbol | [required] A `string` representing asset ledger symbol.
 | assetLedgerData.uriPrefix | [required] A `string` representing prefix of asset URI.
 | assetLedgerData.uriPostfix | [required] A `string` representing postfix of asset URI.
 | expiration | [required] An `integer` number representing the timestamp in milliseconds after which the order expires and can not be performed any more.
 | kind | [required] An `integer` number that equals to `OrderKind.ASSET_LEDGER_DEPLOY_ORDER`.
-| makerId | [required] A `string` representing an Ethereum account address which makes the order. It defaults to the `accountId` of a provider.
+| makerId | [required] A `string` representing an Wanchain account address which makes the order. It defaults to the `accountId` of a provider.
```
Comment (typwrtr) [inline docs/api/wanchain.md]: | makerId | [required] A `string` representing a Wanchain account address which makes the order. It defaults to the `accountId` of a provider.

File: docs/api/wanchain.md [hunk 23]
```diff
@@ -2079,9 +2126,9 @@ This order kind can perform multiple operations such as value transfer, asset tr
 | signature | [required] A `string` representing order signature created by the maker.
 | order.actions | [required] An `array` of [actions order action objects](#actions order-actions).
 | order.expiration | [required] An `integer` number representing the timestamp in milliseconds after which the order expires and can not be performed any more.
-| order.makerId | [required] A `string` representing an Ethereum account address which makes the order. It defaults to the `accountId` of a provider.
+| order.makerId | [required] A `string` representing an Wanchain account address which makes the order. It defaults to the `accountId` of a provider.
```
Comment (typwrtr) [inline docs/api/wanchain.md]: | order.makerId | [required] A `string` representing a Wanchain account address which makes the order. It defaults to the `accountId` of a provider.

File: docs/api/wanchain.md [hunk 24]
```diff
@@ -2091,15 +2138,15 @@ This order kind is used for delegating `ValueLedger` deploy.
 |-|-
 | expiration | [required] An `integer` number representing the timestamp in milliseconds after which the order expires and can not be performed any more.
 | kind | [required] An `integer` number that equals to `OrderKind.ASSET_LEDGER_DEPLOY_ORDER`.
-| makerId | [required] A `string` representing an Ethereum account address which makes the order. It defaults to the `accountId` of a provider.
+| makerId | [required] A `string` representing an Wanchain account address which makes the order. It defaults to the `accountId` of a provider.
```
Comment (typwrtr) [inline docs/api/wanchain.md]: | makerId | [required] A `string` representing a Wanchain account address which makes the order. It defaults to the `accountId` of a provider.

File: docs/api/wanchain.md [hunk 25]
```diff
@@ -2113,13 +2160,18 @@ Multi-order actions define the atomic operations of the actions order.
 |-|-|-
 | CREATE_ASSET | 1 | Create a new asset.
 | UPDATE_ASSET_IMPRINT | 4 | Update asset imprint.
+| SET_ABILITIES | 5 | Sets abilities.
```
Comment (typwrtr) [inline docs/api/wanchain.md]: | SET_ABILITIES | 5 | Set abilities.

File: docs/api/wanchain.md [hunk 26]
```diff
@@ -2113,13 +2160,18 @@ Multi-order actions define the atomic operations of the actions order.
...
 | TRANSFER_ASSET | 2 | Transfer an asset.
 | TRANSFER_VALUE | 3 | Transfer a value.
 
 ::: Warning
 There is a possibility of unintentional behavior where asset imprint can be overwritten if more than one `UPDATE_ASSET_IMPRINT` order per asset is active. Be aware of this when implementing.
 :::
 
+::: Warning
+There is a possibility of unintentional behavior where account abilitites can be overwritten if more than one `SET_ABILITIES` order per account is active. Be aware of this when implementing.
```
Comment (typwrtr) [inline docs/api/wanchain.md]: There is a possibility of unintentional behavior where account abilities could be overwritten if more than one `SET_ABILITIES` order per account is active. Be aware of this when implementing.

File: docs/api/ethereum.md (hunk 1 above)
Comment (MoMannn) [reply docs/api/ethereum.md]: This is code for yellow warning label it has to be lowercase.

**Pull request merged**